packages/components/icons/src/leading-icon/leading-icon.tsx

```diff
@@ -0,0 +1,197 @@
+  color = 'neutral',
+  isInverted = false,
+}: {
+  color: TLeadingIconProps['color'] | 'customSvg';
+  isInverted: TLeadingIconProps['isInverted'];
+}) {
+  switch (color) {
+    case 'white':
+      return isInverted
+        ? {
+            background: designTokens.colorSolid,
+            fill: designTokens.colorNeutral90,
+          }
+        : {
+            background: designTokens.colorSurface,
+            border: `solid ${designTokens.borderWidth1} ${designTokens.colorNeutral90}`,
+            fill: designTokens.colorNeutral60,
+          };
+    case 'neutral':
+    default:
+      return isInverted
+        ? {
+            background: designTokens.colorNeutral40,
+            fill: designTokens.colorNeutral90,
+          }
+        : {
+            background: designTokens.colorNeutral90,
+            fill: designTokens.colorNeutral40,
+          };
+    case 'purple':
+      return isInverted
+        ? {
+            background: designTokens.colorPurple50,
+            fill: designTokens.colorNeutral90,
+          }
+        : {
+            background: designTokens.colorPurple90,
+            fill: designTokens.colorPurple50,
+          };
+    case 'turquoise':
+      return isInverted
+        ? {
+            background: designTokens.colorTurquoise50,
+            fill: designTokens.colorNeutral90,
+          }
+        : {
+            background: designTokens.colorTurquoise90,
+            fill: designTokens.colorTurquoise50,
+          };
+    case 'accent':
+      return isInverted
+        ? {
+            background: designTokens.colorAccent40,
+            fill: designTokens.colorNeutral90,
+          }
+        : {
+            fill: designTokens.colorAccent40,
+            background: designTokens.colorAccent90,
+          };
+    case 'brown':
+      return isInverted
+        ? {
+            background: designTokens.colorBrown50,
+            fill: designTokens.colorNeutral90,
+          }
+        : {
+            background: designTokens.colorBrown90,
+            fill: designTokens.colorBrown50,
+          };
+    case 'customSvg':
+      return {
+        background: designTokens.colorSurface,
+        border: `solid ${designTokens.borderWidth1} ${designTokens.colorNeutral90}`,
+        fill: null,
+      };
+  }
+}
+
+const defaultProps: Pick<TLeadingIconProps, 'color' | 'size' | 'isInverted'> = {
+  color: 'neutral',
+  size: '20',
+  isInverted: false,
+};
+
+const LeadingIcon = (props: TLeadingIconProps) => {
+  if (!props.svg && !props.icon) {
+    warning(
+      !props.icon && !props.svg,
+      'LeadingIcon: you must pass either an icon or svg prop to the LeadingIcon component'
+    );
+    return null;
+  }
+
+  const sizeStyles = getSizeStyles(props.size);
+  const colorStyles = getColorStyles({
+    color: props.svg ? 'customSvg' : props.color,
+    isInverted: props.isInverted,
+  });
+  return (
+    <div
+      // https://emotion.sh/docs/best-practices#use-the-style-prop-for-dynamic-styles
```

Comment:
This is one place where I am going to push back a little bit. I definitely understand the need for consistency.

That being said, this is a component library that is used extensively, and I am going to argue that doing as much as is possible to reduce the amount of 
1) js that needs to be executed to put styles on the page at runtime, and 
2) the amount of css that gets added to the CSSOM by that js,
is an overall win.  

This implementation makes sure that this component generates a single class, instead of generating up to as many classes as there are color themes * sizes (plus more if theres custom svg).

I think that the benefits, along with the fact that this is the approach that is recommended by the emotion docs for this use-case, makes a pretty compelling argument for allowing this pattern at the cost of being less consistent.

Reply:
Thanks for your explanation, @ByronDWall 

I totally understand your point but to me consistency is also a very high priority goal for such a big (and growing) components library in order to keep maintainability in place.

I would buy into taking this as a checkpoint for us to think about how are we are writing the styles and maybe come up with a new best practice and a plan for migrating to it.

My reasoning for not doing so it two folded:
* We are already studying alternatives to `styled-component` as our main library and event if we want to keep using CSS-inJS, so I don't feel this is like the right moment to raise this debate.
* I understand that it's difficult to find the balance between performance and maintainability and also that we all might have different perspectives. Mine, as someone who had to maintain large codebases for many years, I would prefer maintainability to keep codebase manageable and scalable. I'm not saying at all that performance is not important, but I also think we should wight how much benefit we would be gaining and compare it with the code complexity.

Hope this reasoning can make sense to you.

Please feel free to ping me if you prefer to have an online discussion about it 👍

Reply:
@CarlosCortizasCT Your reasoning certainly makes sense, and I understand your concern.  You work in this library every day, Im just visiting - so I'm not going to push back too hard, I just wanted to explicitly state my concerns.

I will change the component up to be consistent in how it handles styles.

I also want to add a +1 for maybe-not-css-in-js when that conversation is had.

Reply:
Thanks for your understanding, @ByronDWall 

Please, do not forget to that a look at [this proposal](https://github.com/commercetools/ui-kit/pull/2708/) to see if it could fit the comments 🙏